<?xml version="1.0" encoding="utf-8" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
    "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
     <head>
<!-- automatically generated by Darwin
prepared on Mon May 16 14:13:31 CEST 2011
running on cbrg-mac-02.ethz.ch
by user zollers
-->
<title>OMA stand-alone</title>
<link href="browser.css" rel="stylesheet" type="text/css"></link>
<link href="manual.css" rel="stylesheet" type="text/css"></link>
</head><body >
<div class="content"><div class="title">OMA stand-alone</div><h1>Introduction</h1>
<p>You can download and install OMA as a stand-alone
            version. Included are the algorithms for OMA itself plus its 
            addition ESPRIT. The software can be installed on Linux (x86, 
            both 64bit and 32bit) and MacOSX (x86, both 32bit and 64bit).</p>
<p>For more information about OMA and ESPRIT in general, 
            please have a look at the OMA browser page:</p>
<p><a href="http://omabrowser.org/Algorithm.html">http://omabrowser.org/Algorithm.html</a></p>
<p>If you have specific questions about the installation 
            or the usage of OMA, please contact 
            {adrian or cdessimoz}@inf.ethz.ch .</p>
<h1>Downloads</h1>
<p>The current version of OMA stand-alone can be 
                found here:</p>
<p><a href="http://omabrowser.org/standalone/OMA.0.99d-4-g7584b4c.tgz">OMA.0.99d-4-g7584b4c.tgz</a></p>
<h1>Installation</h1>
<p>To install OMA stand-alone on your system, download
            the installer, untar the package and run the included installer
            script:</p>
<p><pre>curl http://omabrowser.org/standalone/OMA.0.99d-4-g7584b4c.tgz \
    -o OMA.0.99d-4-g7584b4c.tgz
tar xvzf OMA.0.99d-4-g7584b4c.tgz
cd OMA.0.99d-4-g7584b4c
./install.sh /your/install/prefix</pre>
</p>
<p>If you do not choose an install prefix, OMA will be 
            installed in <tt>/usr/local/OMA</tt>
 (for this, you might need 
            to install it using the root account or <tt>sudo</tt>
).</p>
<p>After installation, make sure the <tt>bin</tt>
 folder of OMA is in your <tt>PATH</tt>
 variable, e.g., if
            you are using bash and used <tt>/your/install/prefix</tt>
 as
            installer prefix, add a line in <tt>~/.profile</tt>
 such as:</p>
<p><pre>export PATH=$PATH:/your/install/prefix/OMA/bin</pre>
</p>
<p>For other shells, choose the appropriate syntax.</p>
<h1>Usage</h1>
<p>First, set up a working directory. Copy the file <tt>parameters.drw</tt>
 into this folder and change it to your needs. 
        Create a directory <tt>DB</tt>
 in your working directory that holds 
        the genome data in FASTA format (see 'File formats') and copy your 
        data into this directory. If you want to use ESPRIT, the FASTA file 
        containing the contigs should be called <tt>{YourGenome}.contig.fa</tt>
. Then, simply call <tt>OMA</tt>
 from your working directory to 
        run OMA and/or ESPRIT</p>
<p>If you have not installed OMA yet, use the complete
        path to <tt>bin/oma</tt>
 in the installer folder to start the
        script.</p>
<p>As an example, assume you installed <tt>OMA</tt>
 in <tt>/your/install/prefix</tt>
 and want to use ESPRIT on two genome 
        files and one file with contigs (all in <tt>/home/you/fasta</tt>
, do something like this:</p>
<p><pre># create working directory
mkdir myWorkingDir
cd myWorkingDir
# create DB directory in working directory
mkdir DB
# copy FASTA files into DB directory
cp /home/you/fasta/yourFirstGenomeFile.fa DB/
cp /home/you/fasta/yourSecondGenomeFile.fa DB/
cp /home/you/fasta/yourContigFile.contig.fa DB/
cp /your/install/prefix/OMA/OMA.0.99d-4-g7584b4c/parameters.drw ./
# adjust parameters
vim parameters.drw
# run OMA
OMA</pre>
</p>
<p>To get a first impression of <tt>OMA</tt>
 you 
        could <tt>cd</tt>
 into the <tt>ToyExample</tt>
 directory, 
        have a look at <tt>parameters.drw</tt>
 and run <tt>OMA</tt>
 to process our example files.</p>
<h1>File Formats</h1>
<h2>Input Files</h2>
<p>OMA uses two different input formats: FASTA files for 
            genome input and a Darwin file for parameter input.</p>
<p>The Fasta format is explained in detail on <a href="http://en.wikipedia.org/wiki/FASTA_format">wikipedia</a>.</p>
<p>OMA uses the greater-than symbol '>' to 
            distinguish labels from sequences (in contrast to the possibility
            of using a semicolon ';'). Each sequence in an MSA is supposed
            to have its own label. Have a look at the FASTA files included in 
            ToyExample/DB in our installer package for some example files.</p>
<p>If you want to use ESPRIT, make sure that FASTA files
            containing contigs are called <tt>{YourGenome}.contig.fa</tt>
.
            So if you want to experiment with some mouse genome, call the 
            FASTA file <tt>mouse.contig.fa</tt>
 or <tt>mymouse.contig.fa</tt>
 or something similar.</p>
<p>Parameter files use Darwin syntax.  Key-value-pairs 
            are written as</p>
<p><pre>key := value;</pre>
</p>
<p>Note the colon in <tt>:=</tt>
 and the semicolon at 
            the end of the line. If your parameter file does not use valid 
            Darwin syntax, OMA will print out a short message and stop its 
            execution.</p>
<h2>Output Files</h2>
<h3>OMA Output</h3>
<p>The output of OMA gets written to files stored in 
            a folder <tt>Output</tt>
 in your working directory. There are 
            three text files plus an additional folder <tt>PairwiseOrthologs</tt>
 that contains one file for each pair of
            your genome sets.</p>
<p>The textfiles are organized as described in Table 1.</p>
<table><caption>Table 1: Contents of the OMA output files</caption><thead>
<tr><th align="left"><b>Filename</b>
</th><th align="left"><b>Contents</b>
</th></tr>
</thead>
<tbody><tr><td colspan=2 style="padding: 0px; border-bottom:1px solid black"></td></tr>
<tr><td align="left"><tt>Map-SeqNum-ID.txt</tt>
</td><td align="left"><p>Lists all genes of all datasets  
                    with their unique sequence number and the labels read from
                    the FASTA files.</p>
</td></tr>
<tr><td align="left"><tt>OrthologousGroups.txt</tt>
</td><td align="left"><p>The groups of orthologs are
                    given as one per row, starting with a unique group
                    identifier, followed by all group members, all separated
                    by tabs.</p>
</td></tr>
<tr><td align="left"><tt>OrthologousMatrix.txt</tt>
</td><td align="left"><p>More compact version of <tt>OrthologousGroups.txt</tt>
. The groups of orthologs
                    are given as matrix with group per row and one genome
                    per tab-separated column. Numbers refer to entry number as
                    listed in the file <tt>Map-SeqNum-ID.txt</tt>
.</p>
</td></tr>
</tbody>
</table>
<p>The textfiles in <tt>Output/PairwiseOrthologs</tt>
 are named according to <tt>{genome a}-{genome b}.txt</tt>
 and consist of a list of pairwise orthologs for the two given
            genomes. Every pair is listed only once, and in no particular
            order. Each line in the file contains one pair; all fields are 
            separated by tabs. In the first two field, the unique IDs of the
            proteins are given. The next two fields contain the labels of
            the proteins, and in the last two fields, the type of orthology
            and (if any) the OMA group is given.</p>
<h3>ESPRIT Output</h3>
<p>ESPRIT stores its output files in a folder<tt>EspritOutput</tt>
 in your working directory. The output
            consists of three text files and one tarball. In the tarball,
            FASTA files with the MSAs of the hits ESPRIT found are stored.
             The other three files are explained in detail in Table 2.</p>
<table><caption>Table 2: Contents of the ESPRIT output files</caption><thead>
<tr><th align="left"><b>Filename</b>
</th><th align="left"><b>Contents</b>
</th></tr>
</thead>
<tbody><tr><td colspan=2 style="padding: 0px; border-bottom:1px solid black"></td></tr>
<tr><td align="left"><tt>params.txt</tt>
</td><td align="left"><p>This file is kept as a reference and 
                    contains all parameters used in the current run.</p>
</td></tr>
<tr><td align="left"><tt>hits.txt</tt>
</td><td align="left"><p>All hits found by ESPRIT are listed in this
                    file. It is a list of contigs, ordered according to their
                    position relative to the putative ortholog. Each line 
                    describes one contig, the fields are separated by tabs.
                    In the first field, the fragment pair ID is printed; the
                    next two fields contain the labels of the first and
                    second fragments found in this hit. The forth and fifth 
                    fields contain the label of the corresponding full gene
                    and its genome name. Then follows the distance difference
                    between the two fragments and the number of positions
                    between them (i.e. the gap); at last, an array is listed
                    containing the IDs of all s3 genes corresponding to this
                    hit.</p>
</td></tr>
<tr><td align="left"><tt>dubious.txt</tt>
</td><td align="left"><p>ESPRIT often detects more candidate pairs than
                    it will list in the <tt>hits.txt</tt>
 file, but not all
                    of them survive the quality check. Still, if you want to
                    see which triplets have been filtered out, have a look at <tt>dubious.txt</tt>
 where they are still listed. The file
                    format is the same as for <tt>hits.txt</tt>
.</p>
</td></tr>
</tbody>
</table>
<h1>Parameters</h1>
<p>All parameters for OMA and/or ESPRIT are
                set in a parameters file. There is an example file in the
                OMA installer package; we encourage you to copy this file
                into your working directory and change it to your needs.</p>
<p>The parameter file consists of two main parts:
                First, general parameters for OMA are set; see Table 3 for
                detailed explanations. Second, more specific parameters 
                that only affect the ESPRIT algorithm can be changed. 
                These parameters are explained in Table 4.
                Note that changing the ESPRIT parameters
                will not have an effect unless you set the boolean variable <tt>UseEsprit</tt>
 to <tt>true</tt>
.</p>
<table><caption>Table 3: General parameters in OMA</caption><thead>
<tr><th align="left"><b>Parameter</b>
</th><th align="left"><b>Meaning</b>
</th><th align="left"><b>Default</b>
</th></tr>
</thead>
<tbody><tr><td colspan=3 style="padding: 0px; border-bottom:1px solid black"></td></tr>
<tr><td align="left"><tt>ReuseCachedResults</tt>
</td><td align="left"><p>If you want to recompute everything from
                        scratch everytime the script is run, set this to <tt>false</tt>
.</p>
</td><td align="left"><tt>true</tt>
</td></tr>
<tr><td align="left"><tt>MinScore</tt>
</td><td align="left"><p>Alignments which have a score lower than <tt>MinScore</tt>
 will not be considered. The scores
                        are in Gonnet PAM matrices units.</p>
</td><td align="left"><tt>181</tt>
</td></tr>
<tr><td align="left"><tt>LengthTol</tt>
</td><td align="left"><p>Length tolerance ratio. If the length of
                        the effective alignment is less than <tt>LengthTol</tt>
 * min( length(s1), length(s2) ), then 
                        the alignment is not considered.</p>
</td><td align="left"><tt>0.61</tt>
</td></tr>
<tr><td align="left"><tt>StablePairTol</tt>
</td><td align="left"><p>During the stable pair formation, if a
                        pair has a distance provable higher than another pair
                        (i.e. <tt>StablePairTol</tt>
 standard deviations
                        away) then it is discarded.</p>
</td><td align="left"><tt>1.81</tt>
</td></tr>
<tr><td align="left"><tt>VerifiedPairTol</tt>
</td><td align="left"><p>Length tolerance ratio. If the length of
                        the effective alignment is less than <tt>LengthTol</tt>
 * min( length(s1), length(s2) ), then the alignment is
                        not considered.</p>
</td><td align="left"><tt>1.53</tt>
</td></tr>
<tr><td align="left"><tt>MinSeqLen</tt>
</td><td align="left"><p>Any sequence which is less than <tt>MinSeqLen</tt>
 amino acids long in regular genomes
                        is not considered.</p>
</td><td align="left"><tt>50</tt>
</td></tr>
</tbody>
</table>
<table><caption>Table 4: ESPRIT parameters</caption><thead>
<tr><th align="left"><b>Parameter</b>
</th><th align="left"><b>Meaning</b>
</th><th align="left"><b>Default</b>
</th></tr>
</thead>
<tbody><tr><td colspan=3 style="padding: 0px; border-bottom:1px solid black"></td></tr>
<tr><td align="left"><tt>UseEsprit</tt>
</td><td align="left"><p>You can either set this to <tt>true</tt>
, which will enable ESPRIT and shut down the parts of
                        OMA that are not directly needed for ESPRIT, or set it
                        to <tt>false</tt>
 to make no use of ESPRIT at all.</p>
</td><td align="left"><tt>false</tt>
</td></tr>
<tr><td align="left"><tt>DistConfLevel</tt>
</td><td align="left"><p>Confidence level variable for contigs.
                        This is the parameter <tt>tol</tt>
 described in the
                        paper.</p>
</td><td align="left"><tt>2</tt>
</td></tr>
<tr><td align="left"><tt>MinProbContig</tt>
</td><td align="left"><p>Minimal proportion of genomes with which
                        contigs form many:1 BestMatches to consider that we
                        might be dealing with fragments of the same gene.
                        This is the parameter <tt>MinRefGenomes</tt>
 described in the paper, normalized by the total
                        number of reference genomes.</p>
</td><td align="left"><tt>0.4</tt>
</td></tr>
<tr><td align="left"><tt>MaxContigOverlap</tt>
</td><td align="left"><p>Maximum overlap between fragments of
                        same gene from different contigs.</p>
</td><td align="left"><tt>5</tt>
</td></tr>
<tr><td align="left"><tt>MinSeqLenContig</tt>
</td><td align="left"><p>Any sequence which is less than <tt>MinSeqLenContig</tt>
 amino acids long in contigs
                        is not considered.</p>
</td><td align="left"><tt>20</tt>
</td></tr>
<tr><td align="left"><tt>MinBestScore</tt>
</td><td align="left"><p>Minimum best score for BestMatch in
                        scaffold recognition.</p>
</td><td align="left"><tt>250</tt>
</td></tr>
</tbody>
</table>
<h1>License</h1>
<p>OMA is licensed under a Creative Commons 
Attribution-Noncommercial-Share Alike 2.5 License. For more info, please 
consult the following page:</p>
<p><a href="http://creativecommons.org/licenses/by-nc-sa/2.5/ch/">http://creativecommons.org/licenses/by-nc-sa/2.5/ch/</a></p>
<p>In a nutshell, OMA is free for non-commercial use.</p>
</div></body></html>

